massa-network/src/peer_info_database.rs

```diff
@@ -443,6 +443,119 @@ impl PeerInfoDatabase {
         self.update()
     }
 
+    pub async fn whitelist(&mut self, ips: Vec<IpAddr>) -> Result<(), NetworkError> {
```

Comment:
Maybe this part of code is a bit hard to read through given the numerous almost equivalent branching, what about spiting it in small chunks?

Reply:
Any idea on how to do that ?

Reply:
```rust
if old_pt == PeerType::WhiteListed { continue }
// decrease from one indent the following code
```